pkg/resources/task.go

```diff
@@ -2,709 +2,648 @@ package resources
-			case "USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE":
-				fieldParameters["user_task_managed_initial_warehouse_size"] = param.Value
-			case "USER_TASK_TIMEOUT_MS":
-				timeout, err := strconv.ParseInt(param.Value, 10, 64)
-				if err != nil {
-					return err
-				}
-
-				fieldParameters["user_task_timeout_ms"] = timeout
-			case "SUSPEND_TASK_AFTER_NUM_FAILURES":
-				num, err := strconv.ParseInt(param.Value, 10, 64)
-				if err != nil {
-					return err
-				}
-
-				fieldParameters["suspend_task_after_num_failures"] = num
-			default:
-				sessionParameters[param.Key] = param.Value
-			}
-		}
-
-		if err := d.Set("session_parameters", sessionParameters); err != nil {
-			return err
-		}
-
-		for key, value := range fieldParameters {
-			// lintignore:R001
-			err = d.Set(key, value)
-			if err != nil {
-				return err
-			}
-		}
-	}
-
-	return nil
+	return []*schema.ResourceData{d}, nil
 }
 
-// CreateTask implements schema.CreateFunc.
-func CreateTask(d *schema.ResourceData, meta interface{}) error {
+func CreateTask(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
-	ctx := context.Background()
 
 	databaseName := d.Get("database").(string)
 	schemaName := d.Get("schema").(string)
 	name := d.Get("name").(string)
+	id := sdk.NewSchemaObjectIdentifier(databaseName, schemaName, name)
+	req := sdk.NewCreateTaskRequest(id, d.Get("sql_statement").(string))
 
-	sqlStatement := d.Get("sql_statement").(string)
-
-	taskId := sdk.NewSchemaObjectIdentifier(databaseName, schemaName, name)
-	createRequest := sdk.NewCreateTaskRequest(taskId, sqlStatement)
+	tasksToResume := make([]sdk.SchemaObjectIdentifier, 0)
 
-	// Set optionals
 	if v, ok := d.GetOk("warehouse"); ok {
-		warehouseId := sdk.NewAccountObjectIdentifier(v.(string))
-		createRequest.WithWarehouse(*sdk.NewCreateTaskWarehouseRequest().WithWarehouse(warehouseId))
-	}
-
-	if v, ok := d.GetOk("user_task_managed_initial_warehouse_size"); ok {
-		size, err := sdk.ToWarehouseSize(v.(string))
+		warehouseId, err := sdk.ParseAccountObjectIdentifier(v.(string))
 		if err != nil {
-			return err
+			return diag.FromErr(err)
 		}
-		createRequest.WithWarehouse(*sdk.NewCreateTaskWarehouseRequest().WithUserTaskManagedInitialWarehouseSize(size))
+		req.WithWarehouse(*sdk.NewCreateTaskWarehouseRequest().WithWarehouse(warehouseId))
 	}
 
 	if v, ok := d.GetOk("schedule"); ok {
-		createRequest.WithSchedule(v.(string))
+		req.WithSchedule(v.(string))
 	}
 
-	if v, ok := d.GetOk("session_parameters"); ok {
-		sessionParameters, err := sdk.GetSessionParametersFrom(v.(map[string]any))
-		if err != nil {
-			return err
-		}
-		createRequest.WithSessionParameters(*sessionParameters)
+	if v, ok := d.GetOk("config"); ok {
+		req.WithConfig(v.(string))
 	}
 
-	if v, ok := d.GetOk("user_task_timeout_ms"); ok {
-		createRequest.WithUserTaskTimeoutMs(v.(int))
+	if v := d.Get("allow_overlapping_execution").(string); v != BooleanDefault {
```

Comment:
did you consider using the helper methods here (check the user resource) - it makes some of this code more compact

Reply:
Yeah, I was going over user resource to look up some of the helper function you added there. Some of them were using this XProperty structs, so I just went the old way for now. I'll try to see what can be still converted in the next pr. Added to the list.

Reply:
You can add new ones! :)

Reply:
(yes, it can be in the next PR)

Reply:
[Adjusted a bit in](https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3170): Not everything could be converted, but I think I changed what I could to attribute helpers, also added 2 of them (one for create and one for read).